libs/locale.cmake

```diff
@@ -98,6 +100,7 @@ if(BOOST_LOCALE_ENABLE_ICONV_BACKEND AND ICONV_FOUND)
     Iconv::Iconv
   )
   target_compile_definitions(Boost_locale_deps INTERFACE BOOST_LOCALE_WITH_ICONV=1)
+  set_property(GLOBAL APPEND PROPERTY Boost_Find_Package Iconv)
```

Comment:
Ah, I see now. This means you aren't searching for boost components but actual third party dependencies. In that case yes, you should most definitely be using `find_dependency()`!

Reply:
The problem with `find_dependency()` is that it will abort the whole `find_package()` call if any of those is not met. I can easily imagine people providing binaries built with MPI enabled and later on users without MPI.
Using this, the other libraries can still be used.

Reply:
I think the point of it is to make sure that `EXACT` is respected, but I've never personally used it. For example, if a boost component is listed under `OPTIONAL_COMPONENTS`, if it finds that boost component but *not* its dependency, that should be a "soft" failure, i.e. treat that component as not found, but keep going. Not sure if this case is handled here, but it's something worth testing (if you haven't already). Honestly it's pretty complicated, I'm not sure what the de facto behavior should be.